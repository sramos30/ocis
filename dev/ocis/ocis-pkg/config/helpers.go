package config

import (
	"io/fs"
	"os"
	"path/filepath"
	"strings"

	gofig "github.com/gookit/config/v2"
	gooyaml "github.com/gookit/config/v2/yaml"
)

var (
	defaultLocations = []string{
		filepath.Join(os.Getenv("HOME"), "/.ocis/config/"),
		"/etc/ocis/",
		".config/",
	}

	// supportedExtensions is determined by gookit/config.
	supportedExtensions = []string{
		"yaml",
		"yml",
	}
)

// DefaultConfigSources returns a slice with matched expected config files. It sugars coat several aspects of config file
// management by assuming there are 3 default locations a config file could be.
// It uses globbing to match a config file by name, and retrieve any supported extension supported by our drivers.
// It sanitizes the output depending on the list of drivers provided.
func DefaultConfigSources(filename string, drivers []string) []string {
	var sources []string

	locations := []string{}
	if v := os.Getenv("OCIS_CONFIG_DIR"); v != "" {
	    locations = append(locations, v)
		// only use the configured config dir
		locations = append(locations, os.Getenv("OCIS_CONFIG_DIR"))
	} else {
		// merge config from all default locations
		locations = append(locations, defaultLocations...)
	}

	for i := range locations {
		dirFS := os.DirFS(locations[i])
		pattern := filename + ".*"
		matched, _ := fs.Glob(dirFS, pattern)
		if len(matched) > 0 {
			// prepend path to results
			for j := 0; j < len(matched); j++ {
				matched[j] = filepath.Join(locations[i], matched[j])
			}
		}
		sources = append(sources, matched...)
	}

	return sanitizeExtensions(sources, drivers, func(a, b string) bool {
		return strings.HasSuffix(filepath.Base(a), b)
	})
}

// sanitizeExtensions removes elements from "set" which extensions are not in "ext".
func sanitizeExtensions(set []string, ext []string, f func(a, b string) bool) []string {
	var r []string
	for i := 0; i < len(set); i++ {
		for j := 0; j < len(ext); j++ {
			if f(filepath.Base(set[i]), ext[j]) {
				r = append(r, set[i])
			}
		}
	}
	return r
}

// BindSourcesToStructs assigns any config value from a config file / env variable to struct `dst`. Its only purpose
// is to solely modify `dst`, not dealing with the config structs; and do so in a thread safe manner.
func BindSourcesToStructs(extension string, dst interface{}) (*gofig.Config, error) {
	sources := DefaultConfigSources(extension, supportedExtensions)
	cnf := gofig.NewWithOptions(extension)
	cnf.WithOptions(func(options *gofig.Options) {
		options.DecoderConfig.TagName = "yaml"
	})
	cnf.AddDriver(gooyaml.Driver)
	_ = cnf.LoadFiles(sources...)

	err := cnf.BindStruct("", &dst)
	if err != nil {
		return nil, err
	}

	return cnf, nil
}
