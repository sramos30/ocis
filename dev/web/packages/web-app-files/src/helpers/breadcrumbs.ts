import { bus } from 'web-pkg/src/instance'

export interface BreadcrumbItem {
  text: string
  to?: string
  allowContextActions?: boolean
  onClick?: () => void
}

export const breadcrumbsFromPath = (
  currentPath: string,
  resourcePath: string
): BreadcrumbItem[] => {
  const pathSplit = (p = '') => p.split('/').filter(Boolean)
  const current = pathSplit(currentPath)
  const resource = pathSplit(resourcePath)

  return resource.map(
    (text, i) =>
      ({
        allowContextActions: true,
        text,
        to: '/' + [...current].splice(0, current.length - resource.length + i + 1).join('/')
      } as BreadcrumbItem)
  )
}

export const concatBreadcrumbs = (...items: BreadcrumbItem[]): BreadcrumbItem[] => {
  const last = items.pop()

  return [
    ...items,
    {
      allowContextActions: last.allowContextActions,
      text: last.text,
      onClick: () => bus.publish('app.files.list.load')
    }
  ]
}
