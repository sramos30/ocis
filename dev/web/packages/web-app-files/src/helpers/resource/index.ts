export * from './asset'
export * from './filter'
export * from './privatePreviewBlob'
export * from './publicPreviewUrl'
export * from './resource'
export * from './sameResource'
