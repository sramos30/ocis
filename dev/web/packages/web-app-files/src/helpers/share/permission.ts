// dummy to trick gettext string extraction into recognizing strings
const $gettext = (str) => {
  return str
}

export class SharePermission {
  private readonly _key: string
  private readonly _bit: number
  private readonly _label: string

  constructor(key: string, bit: number, label: string) {
    this._key = key
    this._bit = bit
    this._label = label
  }

  get key(): string {
    return this._key
  }

  get bit(): number {
    return this._bit
  }

  get label(): string {
    return $gettext(this._label)
  }

  public enabled(bitmask: number): boolean {
    return (bitmask & this.bit) > 0
  }
}

export abstract class SharePermissions {
  static readonly read = new SharePermission('read', 1, $gettext('Read'))
  static readonly update = new SharePermission('update', 2, $gettext('Update'))
  static readonly create = new SharePermission('create', 4, $gettext('Create'))
  static readonly delete = new SharePermission('delete', 8, $gettext('Delete'))
  static readonly share = new SharePermission('share', 16, $gettext('Share'))

  static permissionsToBitmask(permissions: SharePermission[]): number {
    return (permissions || []).reduce((b: number, p: SharePermission) => b | p.bit, 0)
  }

  static bitmaskToPermissions(bitmask: number): SharePermission[] {
    return [this.read, this.update, this.create, this.delete, this.share].reduce(
      (a: SharePermission[], p: SharePermission) => {
        if (p.enabled(bitmask)) {
          a.push(p)
        }
        return a
      },
      []
    )
  }
}
