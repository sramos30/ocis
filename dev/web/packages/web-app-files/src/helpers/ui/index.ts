export * as fileList from './filesList'
export * as resourceTable from './resourceTable'
