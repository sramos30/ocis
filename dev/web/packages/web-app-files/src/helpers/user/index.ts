export * from './avatarUrl'
export * from './types'
