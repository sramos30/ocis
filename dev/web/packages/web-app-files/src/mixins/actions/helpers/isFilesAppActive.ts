import { activeApp } from 'web-pkg/src/composables'

const isFilesAppActive = (activeApp: string): boolean => {
  // FIXME: we should use this constant but it somehow breaks the unit tests
  // return activeApp === FilesApp.appInfo.id
  return activeApp === 'files'
}

export default {
  computed: {
    $_isFilesAppActive(): boolean {
      return isFilesAppActive(activeApp(this.$route))
    }
  }
}
