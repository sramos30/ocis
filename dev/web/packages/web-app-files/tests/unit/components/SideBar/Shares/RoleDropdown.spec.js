import RoleDropdown from '@files/src/components/SideBar/Shares/RoleDropdown.vue'
import { createLocalVue, mount, shallowMount } from '@vue/test-utils'
import GetTextPlugin from 'vue-gettext'
import Vuex from 'vuex'
import DesignSystem from 'owncloud-design-system'
import { PeopleShareRoles } from '../../../../../src/helpers/share'

const localVue = createLocalVue()
localVue.use(DesignSystem)
localVue.use(Vuex)
localVue.use(GetTextPlugin, {
  translations: 'does-not-matter.json',
  silent: true
})

const selectors = {
  button: '.files-recipient-role-select-btn',
  roles: '.files-recipient-role-drop-list',
  roleButtonPrefix: '#files-recipient-role-drop-btn-',
  customPermissionsDrop: '.files-recipient-custom-permissions-drop'
}

const stubs = {
  'oc-button': true,
  'oc-icon': true
}

const store = new Vuex.Store({
  getters: {
    capabilities: () => {
      return {}
    }
  }
})

// needs differentiation between file and folder type?

describe('RoleDropdown', () => {
  describe('for file shares', () => {
    afterEach(() => {
      jest.clearAllMocks()
    })

    describe('when no existing role is present', () => {
      it.each(['folder', 'file'])(
        'renders a button with invite text if no existing role given for resource type %s',
        (type) => {
          const wrapper = getShallowMountedWrapper({
            existingRole: undefined,
            resourceType: type
          })
          expect(wrapper).toMatchSnapshot()
        }
      )
      it.each(['folder', 'file'])(
        'displays a dropdown with all available roles on button click for resource type %s',
        async (type) => {
          const isFolder = type === 'folder'
          const wrapper = getMountedWrapper({ resourceType: type })
          const button = wrapper.find(selectors.button)
          await button.trigger('click')

          expect(wrapper.find(selectors.roles).isVisible()).toBeTruthy()
          for (const role of PeopleShareRoles.list(isFolder)) {
            expect(wrapper.find(selectors.roleButtonPrefix + role.name).exists()).toBeTruthy()
          }
        }
      )
      describe('custom permissions', () => {
        it.each(['folder', 'file'])(
          'opens custom permissions drop when custom permissions item in the roles gets selected for resource type %s',
          async (type) => {
            const isFolder = type === 'folder'
            const wrapper = getMountedWrapper({ resourceType: type })
            const customPermissionsDrop = wrapper.find(selectors.customPermissionsDrop)
            const showHideMock = jest.fn()
            customPermissionsDrop.vm.show = showHideMock
            wrapper.vm.$refs.rolesDrop.tippy = { hide: showHideMock }

            await wrapper
              .find(selectors.roleButtonPrefix + PeopleShareRoles.custom(isFolder).name)
              .trigger('click')

            expect(showHideMock).toHaveBeenCalled()
          }
        )
      })

      //   it.each(['update', 'create', 'delete', 'share'])(
      //     'displays custom permission %s in the custom permissions drop',
      //     (permission) => {
      //       const wrapper = getWrapper()
      //       expect(
      //         wrapper.find(`[data-testid="files-collaborators-permission-${permission}"]`).exists()
      //       ).toBeTruthy()
      //     }
      //   )
      //   it('closes custom permissions drop when they are confirmed', async () => {
      //     const wrapper = getWrapper()
      //     const permissionsDrop = wrapper.find('[data-testid="files-recipient-custom-permissions-drop"]')

      //     permissionsDrop.vm.hide = jest.fn()

      //     await wrapper
      //       .find('[data-testid="files-recipient-custom-permissions-drop-confirm"]')
      //       .trigger('click')

      //     expect(permissionsDrop.vm.hide).toHaveBeenCalled()
      //   })

      //   it('opens role drop when custom permissions drop is cancelled', async () => {
      //     const wrapper = getWrapper()
      //     const rolesDrop = wrapper.find('[data-testid="files-recipient-roles-drop"]')
      //     const permissionsDrop = wrapper.find('[data-testid="files-recipient-custom-permissions-drop"]')

      //     rolesDrop.vm.show = jest.fn()
      //     permissionsDrop.vm.hide = jest.fn()

      //     await wrapper
      //       .find('[data-testid="files-recipient-custom-permissions-drop-cancel"]')
      //       .trigger('click')

      //     expect(permissionsDrop.vm.hide).toHaveBeenCalled()
      //     expect(rolesDrop.vm.show).toHaveBeenCalled()
      //   })
      it.todo('emits an event upon role selection')
    })
    describe('when an existing role is present', () => {
      it.each(['folder', 'file'])(
        'renders a button with existing role if given for resource type %s',
        (type) => {
          const wrapper = getShallowMountedWrapper({
            existingRole: PeopleShareRoles.list(type === 'folder')[0]
          })
          expect(wrapper).toMatchSnapshot()
        }
      )
      it.todo(
        'displays a dropdown with viewer, editor and custom permissions if no custom permissions had been selected'
      )
      it.todo('displays a dropdown with custom permissions if they already were selected')
      //   it('opens custom permissions drop when custom permissions item in the roles is selected', async () => {
      //     const wrapper = getWrapper()
      //     const advancedPermissionsDrop = wrapper.find(
      //       '[data-testid="files-recipient-custom-permissions-drop"]'
      //     )
      //     const showHideMock = jest.fn()
      //     advancedPermissionsDrop.vm.show = showHideMock
      //     wrapper.vm.$refs.rolesDrop.tippy = { hide: showHideMock }

      //     await wrapper
      //       .find('[data-testid="files-recipient-role-drop-btn-advancedRole"]')
      //       .trigger('click')

      //     expect(showHideMock).toHaveBeenCalledTimes(2)
      //   })
      //   it.each(['update', 'create', 'delete', 'share'])(
      //     'displays custom permission %s in the custom permissions drop',
      //     (permission) => {
      //       const wrapper = getWrapper()
      //       expect(
      //         wrapper.find(`[data-testid="files-collaborators-permission-${permission}"]`).exists()
      //       ).toBeTruthy()
      //     }
      //   )
      //   it('closes custom permissions drop when they are confirmed', async () => {
      //     const wrapper = getWrapper()
      //     const permissionsDrop = wrapper.find('[data-testid="files-recipient-custom-permissions-drop"]')

      //     permissionsDrop.vm.hide = jest.fn()

      //     await wrapper
      //       .find('[data-testid="files-recipient-custom-permissions-drop-confirm"]')
      //       .trigger('click')

      //     expect(permissionsDrop.vm.hide).toHaveBeenCalled()
      //   })

      //   it('opens role drop when custom permissions drop is cancelled', async () => {
      //     const wrapper = getWrapper()
      //     const rolesDrop = wrapper.find('[data-testid="files-recipient-roles-drop"]')
      //     const permissionsDrop = wrapper.find('[data-testid="files-recipient-custom-permissions-drop"]')

      //     rolesDrop.vm.show = jest.fn()
      //     permissionsDrop.vm.hide = jest.fn()

      //     await wrapper
      //       .find('[data-testid="files-recipient-custom-permissions-drop-cancel"]')
      //       .trigger('click')

      //     expect(permissionsDrop.vm.hide).toHaveBeenCalled()
      //     expect(rolesDrop.vm.show).toHaveBeenCalled()
      //   })
      it.todo('emits an event upon role selection')
    })
  })
})

function getMountedWrapper(data) {
  return mount(RoleDropdown, getMountOptions(data))
}

function getShallowMountedWrapper(data) {
  return shallowMount(RoleDropdown, getMountOptions(data))
}

function getMountOptions({ existingRole, shareId, resourceType = 'folder' }) {
  return {
    propsData: {
      resource: getResource({
        filename: resourceType === 'folder' ? 'testfolder' : 'testfile',
        extension: resourceType === 'folder' ? '' : 'jpg',
        type: resourceType
      }),
      existingRole,
      shareId,
      allowSharePermission: true
    },
    store,
    localVue,
    stubs
  }
}

function getResource({ filename = 'testFile', extension = 'txt', type = 'file' }) {
  return {
    id: '4',
    fileId: '4',
    icon: type,
    name: type === 'file' ? `${filename}.${extension}` : filename,
    extension: extension,
    path: type === 'file' ? `/${filename}.${extension}` : `/${filename}`,
    type,
    isFolder: type === 'folder',
    mdate: 'Mon, 12 Jul 2021 11:04:33 GMT',
    size: '163',
    indicators: [],
    permissions: 'RDNVW',
    starred: false,
    etag: '"89128c0e8122002db57bd19c9ec33004"',
    shareTypes: [],
    downloadURL: ''
  }
}
