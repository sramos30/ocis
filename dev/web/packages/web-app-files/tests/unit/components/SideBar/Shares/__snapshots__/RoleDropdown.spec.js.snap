// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoleDropdown for file shares when an existing role is present renders a button with existing role if given for resource type file 1`] = `
<span class="oc-flex oc-flex-middle"><oc-button-stub type="button" size="medium" submit="button" variation="passive" appearance="raw" justifycontent="center" gapsize="none" id="files-collaborators-role-button-new" class="files-recipient-role-select-btn"><span>Viewer</span>
<oc-icon-stub name="arrow-down-s" filltype="fill" accessiblelabel="" type="span" size="medium" variation="passive" color=""></oc-icon-stub>
</oc-button-stub>
<oc-drop-stub dropid="oc-drop-17" toggle="#files-collaborators-role-button-new" position="bottom-start" mode="click" closeonclick="true" paddingsize="remove">
  <oc-list-stub aria-label="Select role for the invitation" class="files-recipient-role-drop-list">
    <li>
      <oc-button-stub type="button" size="medium" submit="button" variation="passive" appearance="raw" justifycontent="space-between" gapsize="medium" id="files-recipient-role-drop-btn-viewer" class="files-recipient-role-drop-btn oc-py-xs oc-px-s selected">
        <role-item-stub role="[object Object]" allowsharepermission="true"></role-item-stub>
        <oc-icon-stub name="check" filltype="fill" accessiblelabel="" type="span" size="medium" variation="passive" color=""></oc-icon-stub>
      </oc-button-stub>
    </li>
    <li>
      <oc-button-stub type="button" size="medium" submit="button" variation="passive" appearance="raw" justifycontent="space-between" gapsize="medium" id="files-recipient-role-drop-btn-editor" class="files-recipient-role-drop-btn oc-py-xs oc-px-s">
        <role-item-stub role="[object Object]" allowsharepermission="true"></role-item-stub>
        <!---->
      </oc-button-stub>
    </li>
    <li>
      <oc-button-stub type="button" size="medium" submit="button" variation="passive" appearance="raw" justifycontent="space-between" gapsize="medium" id="files-recipient-role-drop-btn-custom" class="files-recipient-role-drop-btn oc-py-xs oc-px-s">
        <role-item-stub role="[object Object]" allowsharepermission="true"></role-item-stub>
        <!---->
      </oc-button-stub>
    </li>
  </oc-list-stub>
</oc-drop-stub>
<oc-drop-stub dropid="oc-drop-18" toggle="" position="bottom-start" mode="manual" target="#files-collaborators-role-button-new" paddingsize="small" class="files-recipient-custom-permissions-drop">
  <h4 class="oc-text-bold oc-text-initial">Custom permissions</h4>
  <oc-list-stub class="oc-mb">
    <li class="oc-my-xs">
      <oc-checkbox-stub id="files-collaborators-permission-read" disabled="true" value="[object Object],[object Object]" option="[object Object]" label="Read" size="medium" data-testid="files-collaborators-permission-read" class="oc-mr-xs files-collaborators-permission-checkbox"></oc-checkbox-stub>
    </li>
    <li class="oc-my-xs">
      <oc-checkbox-stub id="files-collaborators-permission-update" value="[object Object],[object Object]" option="[object Object]" label="Update" size="medium" data-testid="files-collaborators-permission-update" class="oc-mr-xs files-collaborators-permission-checkbox"></oc-checkbox-stub>
    </li>
    <li class="oc-my-xs">
      <oc-checkbox-stub id="files-collaborators-permission-create" value="[object Object],[object Object]" option="[object Object]" label="Create" size="medium" data-testid="files-collaborators-permission-create" class="oc-mr-xs files-collaborators-permission-checkbox"></oc-checkbox-stub>
    </li>
    <li class="oc-my-xs">
      <oc-checkbox-stub id="files-collaborators-permission-delete" value="[object Object],[object Object]" option="[object Object]" label="Delete" size="medium" data-testid="files-collaborators-permission-delete" class="oc-mr-xs files-collaborators-permission-checkbox"></oc-checkbox-stub>
    </li>
    <li class="oc-my-xs">
      <oc-checkbox-stub id="files-collaborators-permission-share" value="[object Object],[object Object]" option="[object Object]" label="Share" size="medium" data-testid="files-collaborators-permission-share" class="oc-mr-xs files-collaborators-permission-checkbox"></oc-checkbox-stub>
    </li>
  </oc-list-stub>
  <div class="files-recipient-custom-permissions-drop-cancel-confirm-btns">
    <oc-button-stub type="button" size="small" submit="button" variation="passive" appearance="outline" justifycontent="center" gapsize="medium">Cancel</oc-button-stub>
    <oc-button-stub type="button" size="small" submit="button" variation="primary" appearance="filled" justifycontent="center" gapsize="medium">Apply</oc-button-stub>
  </div>
</oc-drop-stub>
</span>
`;

exports[`RoleDropdown for file shares when an existing role is present renders a button with existing role if given for resource type folder 1`] = `
<span class="oc-flex oc-flex-middle"><oc-button-stub type="button" size="medium" submit="button" variation="passive" appearance="raw" justifycontent="center" gapsize="none" id="files-collaborators-role-button-new" class="files-recipient-role-select-btn"><span>Viewer</span>
<oc-icon-stub name="arrow-down-s" filltype="fill" accessiblelabel="" type="span" size="medium" variation="passive" color=""></oc-icon-stub>
</oc-button-stub>
<oc-drop-stub dropid="oc-drop-15" toggle="#files-collaborators-role-button-new" position="bottom-start" mode="click" closeonclick="true" paddingsize="remove">
  <oc-list-stub aria-label="Select role for the invitation" class="files-recipient-role-drop-list">
    <li>
      <oc-button-stub type="button" size="medium" submit="button" variation="passive" appearance="raw" justifycontent="space-between" gapsize="medium" id="files-recipient-role-drop-btn-viewer" class="files-recipient-role-drop-btn oc-py-xs oc-px-s selected">
        <role-item-stub role="[object Object]" allowsharepermission="true"></role-item-stub>
        <oc-icon-stub name="check" filltype="fill" accessiblelabel="" type="span" size="medium" variation="passive" color=""></oc-icon-stub>
      </oc-button-stub>
    </li>
    <li>
      <oc-button-stub type="button" size="medium" submit="button" variation="passive" appearance="raw" justifycontent="space-between" gapsize="medium" id="files-recipient-role-drop-btn-editor" class="files-recipient-role-drop-btn oc-py-xs oc-px-s">
        <role-item-stub role="[object Object]" allowsharepermission="true"></role-item-stub>
        <!---->
      </oc-button-stub>
    </li>
    <li>
      <oc-button-stub type="button" size="medium" submit="button" variation="passive" appearance="raw" justifycontent="space-between" gapsize="medium" id="files-recipient-role-drop-btn-custom" class="files-recipient-role-drop-btn oc-py-xs oc-px-s">
        <role-item-stub role="[object Object]" allowsharepermission="true"></role-item-stub>
        <!---->
      </oc-button-stub>
    </li>
  </oc-list-stub>
</oc-drop-stub>
<oc-drop-stub dropid="oc-drop-16" toggle="" position="bottom-start" mode="manual" target="#files-collaborators-role-button-new" paddingsize="small" class="files-recipient-custom-permissions-drop">
  <h4 class="oc-text-bold oc-text-initial">Custom permissions</h4>
  <oc-list-stub class="oc-mb">
    <li class="oc-my-xs">
      <oc-checkbox-stub id="files-collaborators-permission-read" disabled="true" value="[object Object],[object Object]" option="[object Object]" label="Read" size="medium" data-testid="files-collaborators-permission-read" class="oc-mr-xs files-collaborators-permission-checkbox"></oc-checkbox-stub>
    </li>
    <li class="oc-my-xs">
      <oc-checkbox-stub id="files-collaborators-permission-update" value="[object Object],[object Object]" option="[object Object]" label="Update" size="medium" data-testid="files-collaborators-permission-update" class="oc-mr-xs files-collaborators-permission-checkbox"></oc-checkbox-stub>
    </li>
    <li class="oc-my-xs">
      <oc-checkbox-stub id="files-collaborators-permission-create" value="[object Object],[object Object]" option="[object Object]" label="Create" size="medium" data-testid="files-collaborators-permission-create" class="oc-mr-xs files-collaborators-permission-checkbox"></oc-checkbox-stub>
    </li>
    <li class="oc-my-xs">
      <oc-checkbox-stub id="files-collaborators-permission-delete" value="[object Object],[object Object]" option="[object Object]" label="Delete" size="medium" data-testid="files-collaborators-permission-delete" class="oc-mr-xs files-collaborators-permission-checkbox"></oc-checkbox-stub>
    </li>
    <li class="oc-my-xs">
      <oc-checkbox-stub id="files-collaborators-permission-share" value="[object Object],[object Object]" option="[object Object]" label="Share" size="medium" data-testid="files-collaborators-permission-share" class="oc-mr-xs files-collaborators-permission-checkbox"></oc-checkbox-stub>
    </li>
  </oc-list-stub>
  <div class="files-recipient-custom-permissions-drop-cancel-confirm-btns">
    <oc-button-stub type="button" size="small" submit="button" variation="passive" appearance="outline" justifycontent="center" gapsize="medium">Cancel</oc-button-stub>
    <oc-button-stub type="button" size="small" submit="button" variation="primary" appearance="filled" justifycontent="center" gapsize="medium">Apply</oc-button-stub>
  </div>
</oc-drop-stub>
</span>
`;

exports[`RoleDropdown for file shares when no existing role is present renders a button with invite text if no existing role given for resource type file 1`] = `
<span class="oc-flex oc-flex-middle"><oc-button-stub type="button" size="medium" submit="button" variation="passive" appearance="raw" justifycontent="center" gapsize="none" id="files-collaborators-role-button-new" class="files-recipient-role-select-btn"><span>Invite as viewer</span>
<oc-icon-stub name="arrow-down-s" filltype="fill" accessiblelabel="" type="span" size="medium" variation="passive" color=""></oc-icon-stub>
</oc-button-stub>
<oc-drop-stub dropid="oc-drop-5" toggle="#files-collaborators-role-button-new" position="bottom-start" mode="click" closeonclick="true" paddingsize="remove">
  <oc-list-stub aria-label="Select role for the invitation" class="files-recipient-role-drop-list">
    <li>
      <oc-button-stub type="button" size="medium" submit="button" variation="passive" appearance="raw" justifycontent="space-between" gapsize="medium" id="files-recipient-role-drop-btn-viewer" class="files-recipient-role-drop-btn oc-py-xs oc-px-s selected">
        <role-item-stub role="[object Object]" allowsharepermission="true"></role-item-stub>
        <oc-icon-stub name="check" filltype="fill" accessiblelabel="" type="span" size="medium" variation="passive" color=""></oc-icon-stub>
      </oc-button-stub>
    </li>
    <li>
      <oc-button-stub type="button" size="medium" submit="button" variation="passive" appearance="raw" justifycontent="space-between" gapsize="medium" id="files-recipient-role-drop-btn-editor" class="files-recipient-role-drop-btn oc-py-xs oc-px-s">
        <role-item-stub role="[object Object]" allowsharepermission="true"></role-item-stub>
        <!---->
      </oc-button-stub>
    </li>
    <li>
      <oc-button-stub type="button" size="medium" submit="button" variation="passive" appearance="raw" justifycontent="space-between" gapsize="medium" id="files-recipient-role-drop-btn-custom" class="files-recipient-role-drop-btn oc-py-xs oc-px-s">
        <role-item-stub role="[object Object]" allowsharepermission="true"></role-item-stub>
        <!---->
      </oc-button-stub>
    </li>
  </oc-list-stub>
</oc-drop-stub>
<oc-drop-stub dropid="oc-drop-6" toggle="" position="bottom-start" mode="manual" target="#files-collaborators-role-button-new" paddingsize="small" class="files-recipient-custom-permissions-drop">
  <h4 class="oc-text-bold oc-text-initial">Custom permissions</h4>
  <oc-list-stub class="oc-mb">
    <li class="oc-my-xs">
      <oc-checkbox-stub id="files-collaborators-permission-read" disabled="true" value="[object Object],[object Object]" option="[object Object]" label="Read" size="medium" data-testid="files-collaborators-permission-read" class="oc-mr-xs files-collaborators-permission-checkbox"></oc-checkbox-stub>
    </li>
    <li class="oc-my-xs">
      <oc-checkbox-stub id="files-collaborators-permission-update" value="[object Object],[object Object]" option="[object Object]" label="Update" size="medium" data-testid="files-collaborators-permission-update" class="oc-mr-xs files-collaborators-permission-checkbox"></oc-checkbox-stub>
    </li>
    <li class="oc-my-xs">
      <oc-checkbox-stub id="files-collaborators-permission-share" value="[object Object],[object Object]" option="[object Object]" label="Share" size="medium" data-testid="files-collaborators-permission-share" class="oc-mr-xs files-collaborators-permission-checkbox"></oc-checkbox-stub>
    </li>
  </oc-list-stub>
  <div class="files-recipient-custom-permissions-drop-cancel-confirm-btns">
    <oc-button-stub type="button" size="small" submit="button" variation="passive" appearance="outline" justifycontent="center" gapsize="medium">Cancel</oc-button-stub>
    <oc-button-stub type="button" size="small" submit="button" variation="primary" appearance="filled" justifycontent="center" gapsize="medium">Apply</oc-button-stub>
  </div>
</oc-drop-stub>
</span>
`;

exports[`RoleDropdown for file shares when no existing role is present renders a button with invite text if no existing role given for resource type folder 1`] = `
<span class="oc-flex oc-flex-middle"><oc-button-stub type="button" size="medium" submit="button" variation="passive" appearance="raw" justifycontent="center" gapsize="none" id="files-collaborators-role-button-new" class="files-recipient-role-select-btn"><span>Invite as viewer</span>
<oc-icon-stub name="arrow-down-s" filltype="fill" accessiblelabel="" type="span" size="medium" variation="passive" color=""></oc-icon-stub>
</oc-button-stub>
<oc-drop-stub dropid="oc-drop-3" toggle="#files-collaborators-role-button-new" position="bottom-start" mode="click" closeonclick="true" paddingsize="remove">
  <oc-list-stub aria-label="Select role for the invitation" class="files-recipient-role-drop-list">
    <li>
      <oc-button-stub type="button" size="medium" submit="button" variation="passive" appearance="raw" justifycontent="space-between" gapsize="medium" id="files-recipient-role-drop-btn-viewer" class="files-recipient-role-drop-btn oc-py-xs oc-px-s selected">
        <role-item-stub role="[object Object]" allowsharepermission="true"></role-item-stub>
        <oc-icon-stub name="check" filltype="fill" accessiblelabel="" type="span" size="medium" variation="passive" color=""></oc-icon-stub>
      </oc-button-stub>
    </li>
    <li>
      <oc-button-stub type="button" size="medium" submit="button" variation="passive" appearance="raw" justifycontent="space-between" gapsize="medium" id="files-recipient-role-drop-btn-editor" class="files-recipient-role-drop-btn oc-py-xs oc-px-s">
        <role-item-stub role="[object Object]" allowsharepermission="true"></role-item-stub>
        <!---->
      </oc-button-stub>
    </li>
    <li>
      <oc-button-stub type="button" size="medium" submit="button" variation="passive" appearance="raw" justifycontent="space-between" gapsize="medium" id="files-recipient-role-drop-btn-custom" class="files-recipient-role-drop-btn oc-py-xs oc-px-s">
        <role-item-stub role="[object Object]" allowsharepermission="true"></role-item-stub>
        <!---->
      </oc-button-stub>
    </li>
  </oc-list-stub>
</oc-drop-stub>
<oc-drop-stub dropid="oc-drop-4" toggle="" position="bottom-start" mode="manual" target="#files-collaborators-role-button-new" paddingsize="small" class="files-recipient-custom-permissions-drop">
  <h4 class="oc-text-bold oc-text-initial">Custom permissions</h4>
  <oc-list-stub class="oc-mb">
    <li class="oc-my-xs">
      <oc-checkbox-stub id="files-collaborators-permission-read" disabled="true" value="[object Object],[object Object]" option="[object Object]" label="Read" size="medium" data-testid="files-collaborators-permission-read" class="oc-mr-xs files-collaborators-permission-checkbox"></oc-checkbox-stub>
    </li>
    <li class="oc-my-xs">
      <oc-checkbox-stub id="files-collaborators-permission-update" value="[object Object],[object Object]" option="[object Object]" label="Update" size="medium" data-testid="files-collaborators-permission-update" class="oc-mr-xs files-collaborators-permission-checkbox"></oc-checkbox-stub>
    </li>
    <li class="oc-my-xs">
      <oc-checkbox-stub id="files-collaborators-permission-create" value="[object Object],[object Object]" option="[object Object]" label="Create" size="medium" data-testid="files-collaborators-permission-create" class="oc-mr-xs files-collaborators-permission-checkbox"></oc-checkbox-stub>
    </li>
    <li class="oc-my-xs">
      <oc-checkbox-stub id="files-collaborators-permission-delete" value="[object Object],[object Object]" option="[object Object]" label="Delete" size="medium" data-testid="files-collaborators-permission-delete" class="oc-mr-xs files-collaborators-permission-checkbox"></oc-checkbox-stub>
    </li>
    <li class="oc-my-xs">
      <oc-checkbox-stub id="files-collaborators-permission-share" value="[object Object],[object Object]" option="[object Object]" label="Share" size="medium" data-testid="files-collaborators-permission-share" class="oc-mr-xs files-collaborators-permission-checkbox"></oc-checkbox-stub>
    </li>
  </oc-list-stub>
  <div class="files-recipient-custom-permissions-drop-cancel-confirm-btns">
    <oc-button-stub type="button" size="small" submit="button" variation="passive" appearance="outline" justifycontent="center" gapsize="medium">Cancel</oc-button-stub>
    <oc-button-stub type="button" size="small" submit="button" variation="primary" appearance="filled" justifycontent="center" gapsize="medium">Apply</oc-button-stub>
  </div>
</oc-drop-stub>
</span>
`;
